lib/binaries/config_source.ts

```diff
@@ -14,7 +14,8 @@ export abstract class ConfigSource {
   opt_ignoreSSL: boolean;
   opt_proxy: string;
 
-  abstract getUrl(version: string): Promise<{url: string, version: string}>;
+  abstract getUrl(version: string, opt_proxy?: string, opt_ignoreSSL?: boolean):
```

Comment:
You're not actually using `opt_proxy` or `opt_ignoreSSL` here.  You want to remove the declarations above and do 

```ts
getUrl(version: string, public opt_proxy?: string, public opt_ignoreSSL?: boolean):
```